ceno_zkvm/src/circuit_builder.rs

```diff
@@ -239,6 +243,21 @@ impl<E: ExtensionField> ConstraintSystem<E> {
         wit_in
     }
 
+    pub fn create_structural_witin<NR: Into<String>, N: FnOnce() -> NR>(&mut self, n: N) -> StructuralWitIn{
+        let wit_in = StructuralWitIn {
```

Comment:
I see what you are trying to do, but I think this is a bit easier to read:

```diff
diff --git a/ceno_zkvm/src/circuit_builder.rs b/ceno_zkvm/src/circuit_builder.rs
index 0d19a621..1c61f078 100644
--- a/ceno_zkvm/src/circuit_builder.rs
+++ b/ceno_zkvm/src/circuit_builder.rs
@@ -248,12 +248,9 @@ impl<E: ExtensionField> ConstraintSystem<E> {
         n: N,
     ) -> StructuralWitIn {
         let wit_in = StructuralWitIn {
-            id: {
-                let id = self.num_structural_witin;
-                self.num_structural_witin = self.num_structural_witin.strict_add(1);
-                id
-            },
+            id: self.num_structural_witin,
         };
+        self.num_structural_witin = self.num_structural_witin.strict_add(1);
 
         let path = self.ns.compute_path(n().into());
         self.witin_namespace_map.push(path);
```

Reply:
ok